spec/metadata/linux/LinuxPackages_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'metadata/linux/LinuxPackages'
+
+describe MiqLinux::Packages do
+  let(:fs) { double("MiqFS") }
+
+  before do
+    allow(fs).to receive(:fileExists?).and_return(false)
+  end
+
+  context "with an RPM directory" do
+    before do
+      expect(fs).to receive(:fileDirectory?).with(MiqLinux::Packages::RPM_DB).and_return(true)
+    end
+
+    context "with a Packages Berkeley DB file" do
+      before do
+        expect(fs)
+          .to receive(:fileOpen)
+          .with(File.join(MiqLinux::Packages::RPM_DB, "Packages"), "r")
+          .and_return(File.open(File.expand_path('../../db/MiqBdb/data/rpm/Packages', __dir__), 'r'))
+      end
+
+      it "returns a list of rpm packages" do
+        result = described_class.new(fs)
+
+        expect(result.instance_variable_get(:@packages).count).to eq(690)
```

Comment:
I'm not sure why we don't have an attr_reader for `@packages` here, `toXml` and `toString` are the only "interface" methods but I think access to packages would be handy also